### PR TITLE
Improve appstreams context selection

### PR DESCRIPTION
**Pull request by parlt91**
## What does this PR change?

Fix an issue where parsing appstream modules would from clients would parse all modules not just active ones.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: **Bugfix**

- [x] **DONE**

## Test coverage
- No tests: **Bugfix**

- [x] **DONE**

## Links

Issue(s): https://github.com/SUSE/spacewalk/issues/25484

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

**Comment by cbbayburt** [comment]
Thanks, @parlt91 looks good.

I am going to test it with some corner cases before approving though.

**Review by cbbayburt — DISMISSED** [review]
There's still no Rocky 8 update available since the last time, so I cannot see if context values change, but I tested everything else and it works as expected.

In the rare case that a context value changes with a module update, (RH-like 8 only) we might still have problems suggesting the correct package for install. There's no easy fix for that, but at least the workaround would be to install such packages from the CLI directly. Hopefully, we won't encounter this in real life.

Thanks for the fix :+1:

File: susemanager-utils/susemanager-sls/src/modules/appstreams.py [hunk 1]
```diff
@@ -92,6 +92,26 @@ def _get_module_info(module_names):
         return []
 
     module_info_output = result.stdout.splitlines()
+    active_modules = []
```
Comment (meaksh) [inline susemanager-utils/susemanager-sls/src/modules/appstreams.py:95]: We should also change this: https://github.com/uyuni-project/uyuni/pull/9372/files#diff-0afe89c35f776dab892a3a79e2f20c5db733ec5c7c2002251a6b205145e7bf43L86 as now the function get the info for active modules only.

Also, before starting parsing the output, I would add an example output to understand better what the loop is trying parse/achieve.

File: susemanager-utils/susemanager-sls/src/modules/appstreams.py [hunk 2]
```diff
@@ -92,6 +92,26 @@ def _get_module_info(module_names):
...
+            )
+            if end != -1:
+                active_modules+=module_info_output[start-1:end+start]
+                module_info_output = module_info_output[end+start:-1]
+            else:
+                active_modules+=module_info_output[start-1:end]
+                module_info_output=active_modules
+                break
+        else:
+            break
```
Comment (meaksh) [inline susemanager-utils/susemanager-sls/src/modules/appstreams.py:114]: I would probably rather prefer to iterate `module_info_output` outside of a `while True` loop, for example, iterating `module_info_output` line-by-line (if that makes sense), and then filtering and collecting only the content of active modules we want.

File: susemanager-utils/susemanager-sls/src/modules/appstreams.py [hunk 3]
```diff
@@ -92,6 +92,26 @@ def _get_module_info(module_names):
...
+            end = next(
+                (i for i, e in enumerate(module_info_output[start:-1])
+                if e.startswith("Name")), -1
+            )
+            if end != -1:
+                active_modules+=module_info_output[start-1:end+start]
+                module_info_output = module_info_output[end+start:-1]
+            else:
+                active_modules+=module_info_output[start-1:end]
+                module_info_output=active_modules
```
Comment (meaksh) [inline susemanager-utils/susemanager-sls/src/modules/appstreams.py:111]: Styling nitpick:

```suggestion
                (i for i, e in enumerate(module_info_output[start : -1])
                if e.startswith("Name")), -1
            )
            if end != -1:
                active_modules += module_info_output[start - 1 : end + start]
                module_info_output = module_info_output[end + start : -1]
            else:
                active_modules += module_info_output[start - 1 : end]
                module_info_output = active_modules
```

File: susemanager-utils/susemanager-sls/src/modules/appstreams.py (hunk 2 above)
Comment (meaksh) [reply susemanager-utils/susemanager-sls/src/modules/appstreams.py]: Sorry for the bad explanation! I was thinking on something a bit more easy to follow, maybe using "regex", like:

```python
    module_info_output = []
    for module in re.findall(r"(Name\s+:.*?)(?=\n\s*\n|$)", result.stdout, re.DOTALL):
        if re.search(r"Stream\s+:.*\[a\]", module):
            module_info_output += module.splitlines()
```

**Review by meaksh — APPROVED** [review]
LGTM, thanks!
